src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -187,4 +188,29 @@ public void addSameStringIdTwiceThrowsKeyCollisionException() {
         database.addString(string);
         fail();
     }
+
+    @Test
+    public void testAddEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry();
+        TestEventListener tel = new TestEventListener();
+        database.registerListener(tel);
+        database.insertEntry(shouldBeEntry);
+        BibEntry isEntry = tel.getBibEntry();
+        assertEquals(shouldBeEntry, isEntry);
+    }
+
+    @Test
+    public void testRemoveEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry();
+        TestEventListener tel = new TestEventListener();
+        database.registerListener(tel);
+        database.insertEntry(shouldBeEntry);
+        tel = new TestEventListener();
```

Comment:
Yes, but this problem can also be circumvented by calling the methods in this order
````
database.insertEntry(...);
TestEventListener listener = new ...;
database.registerListener()

database.remove()

assertEquals()
````